### ISSUE TITLE
Problem installing Atari 

### ISSUE THREAD

**Issue by ghost** [openai/gym#779]
I'm currently trying to install Atari on my Mac with the following commands:
conda create --name tf-gym --clone tf
source activate tf-gym
pip install gym[atari]

I have Anaconda installed and have tensorflow open for that environment but when I enter the third command to install the gym, it returns the following errors. 

Last login: Mon Nov 20 17:29:50 on ttys000
Jarviss-MacBook-Pro:~ Jarvis$ conda create --name tf-gym --clone tf

CondaValueError: prefix already exists: /anaconda3/envs/tf-gym

Jarviss-MacBook-Pro:~ Jarvis$ source activate tf-gym
(tf-gym) Jarviss-MacBook-Pro:~ Jarvis$ pip install gym[atari]
Collecting gym[atari]
Requirement already satisfied: numpy>=1.10.4 in /anaconda3/envs/tf-gym/lib/python3.6/site-packages (from gym[atari])
Requirement already satisfied: six in /anaconda3/envs/tf-gym/lib/python3.6/site-packages (from gym[atari])
Requirement already satisfied: pyglet>=1.2.0 in /anaconda3/envs/tf-gym/lib/python3.6/site-packages (from gym[atari])
Requirement already satisfied: requests>=2.0 in /anaconda3/envs/tf-gym/lib/python3.6/site-packages (from gym[atari])
Collecting atari-py>=0.1.1; extra == "atari" (from gym[atari])
  Using cached atari-py-0.1.1.tar.gz
Collecting Pillow; extra == "atari" (from gym[atari])
  Using cached Pillow-4.3.0-cp36-cp36m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting PyOpenGL; extra == "atari" (from gym[atari])
Requirement already satisfied: future in /anaconda3/envs/tf-gym/lib/python3.6/site-packages (from pyglet>=1.2.0->gym[atari])
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /anaconda3/envs/tf-gym/lib/python3.6/site-packages (from requests>=2.0->gym[atari])
Requirement already satisfied: certifi>=2017.4.17 in /anaconda3/envs/tf-gym/lib/python3.6/site-packages (from requests>=2.0->gym[atari])
Requirement already satisfied: urllib3<1.23,>=1.21.1 in /anaconda3/envs/tf-gym/lib/python3.6/site-packages (from requests>=2.0->gym[atari])
Requirement already satisfied: idna<2.7,>=2.5 in /anaconda3/envs/tf-gym/lib/python3.6/site-packages (from requests>=2.0->gym[atari])
Collecting olefile (from Pillow; extra == "atari"->gym[atari])
Building wheels for collected packages: atari-py
  Running setup.py bdist_wheel for atari-py ... error
  Complete output from command /anaconda3/envs/tf-gym/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/2p/ywwty2x166zcx2_qzhl2460h0000gn/T/pip-build-lkqe_bo1/atari-py/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /var/folders/2p/ywwty2x166zcx2_qzhl2460h0000gn/T/tmpxkc_7u4hpip-wheel- --python-tag cp36:
  running bdist_wheel
  running build
  mkdir -p build && cd build && cmake .. && make -j4
  /bin/sh: cmake: command not found
  make: *** [build] Error 127
  Could not build atari-py: Command '['make', 'build', '-C', 'atari_py/ale_interface', '-j', '7']' returned non-zero exit status 2.. (HINT: are you sure cmake is installed? You might also be missing a library. Atari-py requires: zlib [installable as 'apt-get install zlib1g-dev' on Ubuntu].)
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/private/var/folders/2p/ywwty2x166zcx2_qzhl2460h0000gn/T/pip-build-lkqe_bo1/atari-py/setup.py", line 36, in <module>
      tests_require=['nose2']
    File "/anaconda3/envs/tf-gym/lib/python3.6/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/anaconda3/envs/tf-gym/lib/python3.6/distutils/dist.py", line 955, in run_commands
      self.run_command(cmd)
    File "/anaconda3/envs/tf-gym/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/anaconda3/envs/tf-gym/lib/python3.6/site-packages/wheel/bdist_wheel.py", line 179, in run
      self.run_command('build')
    File "/anaconda3/envs/tf-gym/lib/python3.6/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/anaconda3/envs/tf-gym/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/private/var/folders/2p/ywwty2x166zcx2_qzhl2460h0000gn/T/pip-build-lkqe_bo1/atari-py/setup.py", line 16, in run
      subprocess.check_call(cmd)
    File "/anaconda3/envs/tf-gym/lib/python3.6/subprocess.py", line 291, in check_call
      raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['make', 'build', '-C', 'atari_py/ale_interface', '-j', '7']' returned non-zero exit status 2.
  
  ----------------------------------------
  Failed building wheel for atari-py
  Running setup.py clean for atari-py
Failed to build atari-py
Installing collected packages: atari-py, olefile, Pillow, PyOpenGL, gym
  Running setup.py install for atari-py ... error
    Complete output from command /anaconda3/envs/tf-gym/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/2p/ywwty2x166zcx2_qzhl2460h0000gn/T/pip-build-lkqe_bo1/atari-py/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /var/folders/2p/ywwty2x166zcx2_qzhl2460h0000gn/T/pip-rthr9uw1-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    mkdir -p build && cd build && cmake .. && make -j4
    /bin/sh: cmake: command not found
    make: *** [build] Error 127
    Could not build atari-py: Command '['make', 'build', '-C', 'atari_py/ale_interface', '-j', '7']' returned non-zero exit status 2.. (HINT: are you sure cmake is installed? You might also be missing a library. Atari-py requires: zlib [installable as 'apt-get install zlib1g-dev' on Ubuntu].)
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/2p/ywwty2x166zcx2_qzhl2460h0000gn/T/pip-build-lkqe_bo1/atari-py/setup.py", line 36, in <module>
        tests_require=['nose2']
      File "/anaconda3/envs/tf-gym/lib/python3.6/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/anaconda3/envs/tf-gym/lib/python3.6/distutils/dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "/anaconda3/envs/tf-gym/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/anaconda3/envs/tf-gym/lib/python3.6/site-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/anaconda3/envs/tf-gym/lib/python3.6/distutils/command/install.py", line 545, in run
        self.run_command('build')
      File "/anaconda3/envs/tf-gym/lib/python3.6/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/anaconda3/envs/tf-gym/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/private/var/folders/2p/ywwty2x166zcx2_qzhl2460h0000gn/T/pip-build-lkqe_bo1/atari-py/setup.py", line 16, in run
        subprocess.check_call(cmd)
      File "/anaconda3/envs/tf-gym/lib/python3.6/subprocess.py", line 291, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['make', 'build', '-C', 'atari_py/ale_interface', '-j', '7']' returned non-zero exit status 2.
    
    ----------------------------------------
Command "/anaconda3/envs/tf-gym/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/2p/ywwty2x166zcx2_qzhl2460h0000gn/T/pip-build-lkqe_bo1/atari-py/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /var/folders/2p/ywwty2x166zcx2_qzhl2460h0000gn/T/pip-rthr9uw1-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /private/var/folders/2p/ywwty2x166zcx2_qzhl2460h0000gn/T/pip-build-lkqe_bo1/atari-py/
(tf-gym) Jarviss-MacBook-Pro:~ Jarvis$ 

I've also tried to install gym[all] but then it just returns the same   Failed building wheel for XYZ error for like each package. 

Any help is greatly appreciated.

**Comment by quanvuong** [comment]
Your error trace mentions "/bin/sh: cmake: command not found". Do you have cmake? When you run cmake in terminal, what do you see ?

**Comment by wt2247** [comment]
After running cmake it shows options of path, so shall we specify the path to existing build?

**Comment by Yangzesen** [comment]
I have exact problem on Ubuntu, appreciate any help...

**Comment by FirefoxMetzger** [comment]
I'd say cmake or zlip is the issue here. From your stacktrace:

    /bin/sh: cmake: command not found
    make: *** [build] Error 127
    Could not build atari-py: Command '['make', 'build', '-C', 'atari_py/ale_interface', '-j', '7']' returned non-
    zero exit status 2.. (HINT: are you sure cmake is installed? You might also be missing a library. Atari-py 
    requires: zlib [installable as 'apt-get install zlib1g-dev' on Ubuntu].)

1. try the suggested `sudo apt install zlib1g-dev` (even if that fails it will tell us something)
2. check if cmake is available not just to `/bin/bash` (your terminal), but also to /bin/sh

**Comment by MFajcik** [comment]
Yes as @FirefoxMetzger  says installing cmake and zlib fixes this issue.
On Ubuntu 18
`sudo apt install cmake libz-dev`

**Comment by shuruiz** [comment]
I got exact same issue as yours. For OSX, you can use `brew install cmake` instead to get rid of this issue. The problem is that you need to install cmake first when build the gym[atari].

**Comment by koseoglum** [comment]
It solved my problem on Ubuntu. 

> Yes as @FirefoxMetzger says installing cmake and zlib fixes this issue.
> On Ubuntu 18
> `sudo apt install cmake libz-dev`

**Comment by shashanksira** [comment]
> Yes as @FirefoxMetzger says installing cmake and zlib fixes this issue.
> On Ubuntu 18
> `sudo apt install cmake libz-dev`



> It solved my problem on Ubuntu.
> 
> > Yes as @FirefoxMetzger says installing cmake and zlib fixes this issue.
> > On Ubuntu 18
> > `sudo apt install cmake libz-dev`



> It solved my problem on Ubuntu.
> 
> > Yes as @FirefoxMetzger says installing cmake and zlib fixes this issue.
> > On Ubuntu 18
> > `sudo apt install cmake libz-dev`



> It solved my problem on Ubuntu.
> 
> > Yes as @FirefoxMetzger says installing cmake and zlib fixes this issue.
> > On Ubuntu 18
> > `sudo apt install cmake libz-dev`

Worked for me on ubuntu. I also had installed tensorflow-gpu and cmake prior to this

**Comment by christopherhesse** [comment]
We now release binaries for mac and linux, please file a new issue on `atari-py` if these do not work for you.
